src/SDK/Trace/SpanProcessor/BatchSpanProcessor.php

```diff
@@ -72,6 +75,8 @@ public function __construct(
      */
     public function onStart(ReadWriteSpanInterface $span, ?Context $parentContext = null): void
     {
+        $event = new StartSpanEvent([$span]);
```

Comment:
Why is the span in an array? Wouldn't it make sense for this event to accept a `SpanInterface` (or even `WritableSpanInterface`)? Then it's really clear what you're working with (and IDEs will be happy)

Reply:
- All the Event classes should Ideally Inherit an EventInterface so the getArray(getObject getSpan or whichever name we wanna use for the method) should have the same return type which can be inherited by all the Events. It might be that a particular Event wants to return 'Tracer' then it won't be able be able to implement the interface.
- In some case there may be a requirement of returning more than one object
Considing these 2 cases I thought it was best to return Array. What is your opinion?

Reply:
Have you looked at what other event-based implementations do? Most of my experience with events is with zf2/laminas - they use `target` to refer to the thing the event is operating on (and the type would probably be `mixed`).
It might help you to come up with a solid solution if you can come up with a concrete use-case for having events, and ensure that it can be supported. Once events are implemented, what problem would you like to use events to solve?

Reply:
Good point. In that case, what about type-hinting only in phpdoc, and not adding a php type-hint (which we tidy up in the future when 7.4 support is dropped)

Reply:
I think that `object` is the right type to type-hint to. `mixed` can include things like resources, integers, booleans etc which I wouldn't expect to be the targets of an event